.github/workflows/waspc-ci.yaml

```diff
@@ -90,7 +93,7 @@ jobs:
       - name: Set up Node
         uses: actions/setup-node@v3
         with:
-          node-version: '18'
+          node-version: ${{ matrix.node-version }}
 
       - name: Cache
         uses: actions/cache@v3
```

Comment:
One thing to also consider is cache -> if we were caching any node stuff, like node_modules or something, we would also want to add node version to cache key, but from what I see we are caching only cabal stuff, so no need to add anything, all good!